superset/assets/src/explore/components/controls/MetricsControl.jsx

```diff
@@ -117,8 +117,8 @@ export default class MetricsControl extends React.PureComponent {
 
   componentWillReceiveProps(nextProps) {
     if (
-      this.props.columns !== nextProps.columns ||
-      this.props.savedMetrics !== nextProps.savedMetrics
+      JSON.stringify(this.props.columns) !== JSON.stringify(nextProps.columns) ||
```

Comment:
I don't think `JSON.strinigfy` is good to use for deep equality:
https://stackoverflow.com/questions/15376185/is-it-fine-to-use-json-stringify-for-deep-comparisons-and-cloning

How about lodash's isEqual? https://lodash.com/docs/4.17.10#isEqual